examples/multiplexer.js

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2018-2022, NuoDB, Inc.
+// All rights reserved.
+//
+// Redistribution and use permitted under the terms of the 3-clause BSD license.
+
+const { ShardMultiplexer }= require('..');
+const defaultConnectionConfig = require('../test/config');
+const Pool = require('../lib/pool');
+const MULTIPLEXER_POLL_INTERVAL = 1000;
+const https = require('http');
+
+const getKVStoreValue = (hostname,port,key) => new Promise((res,rej)  => {
+  const options = {
+    hostname,
+    port,
+    path: `/api/1/kvstore/${key}`,
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  }
+
+  // handle request response
+  const req = https.request(options, (response) => {
+    response.on('data', (d) => {
+      const responseMessage = JSON.parse(d.toString('utf8'));
+      res(responseMessage.value);
+    });
+  });
+
+  // handle request error
+  req.on('error', (e) => {
+    rej(e);
+  });
+
+  // send request
+  req.end();
+});
+
+
+const getLatestNumShards = async () => {
+  const HOST = 'localhost';
+  const PORT = '8888';
+  const KEY = 'numShards';
```

Comment:
I would recommend qualifying this further to avoid collisions, for example, by naming the key `nodeJsDriver.multiplexer.numShards` or something.

Reply:
I'm thinking if we parameterize the setup for host/port it might be best practice to also parameterize the key and leave it up to the person running the example to decide

Reply:
I didn't notice the `example/` in the path. Is this like a reference implementation of a working example that the app can use? I don't think the 3DMessaging developers will know enough about how NuoDB works to be able to change anything here, so I think we should "productize" this a bit and make it part of the API.

Reply:
Yes this is meant to be a fit for purpose reference for the 3DMessaging team, altho I do see your point about potential confusion about what value to set the host/port to in a non-local development situation. So I definitely agree that this should be "productized" a bit more for delivery.

However, I'm stuck on how that should happen. I am not sure that I'm sold on the idea that the capacity for querying the admin layer using the nuoadmin REST API should be a part of the multiplexer API. In the sense of maintaining a separation of concerns, the multiplexer is only concerned about providing the utility of managed connection pools, and providing the polling/mapping functionality to let the user define a poll (if desired, it is not necessary to use the multiplexer to poll), and to let the user define a mapping function which will be used when requesting a connection.

I feel like if we add in a fragment of a node.js implementation of the nuoadmin REST API into the Multiplexer we start having the multiplexer doing more things than it is described to do. I'm certainly not opposed to providing a more productized example of using the REST API, or creating a new class for interacting with nuoadmin though. What do you think?